homeassistant/components/flux_led/light.py

```diff
@@ -308,101 +379,103 @@ def effect(self):
-        if not self.is_on:
-            self._bulb.turnOn()
+    @property
+    def device_state_attributes(self):
+        """Return the attributes."""
+        self._attrs["ip_address"] = self._ip_address
+
+        return self._attrs
+
+    @property
+    def device_info(self):
+        """Return the device information."""
+        device_name = "FluxLED/Magic Home"
+        device_model = "LED Lights"
+
+        return {
+            ATTR_IDENTIFIERS: {(DOMAIN, self._unique_id)},
+            ATTR_NAME: self._name,
+            ATTR_MANUFACTURER: device_name,
+            ATTR_MODEL: device_model,
+        }
 
+    async def async_turn_on(self, **kwargs):
+        """Turn on the light."""
+
+        rgb = None
         hs_color = kwargs.get(ATTR_HS_COLOR)
 
         if hs_color:
             rgb = color_util.color_hs_to_RGB(*hs_color)
-        else:
-            rgb = None
 
         brightness = kwargs.get(ATTR_BRIGHTNESS)
         effect = kwargs.get(ATTR_EFFECT)
         white = kwargs.get(ATTR_WHITE_VALUE)
-        color_temp = kwargs.get(ATTR_COLOR_TEMP)
-
-        # handle special modes
-        if color_temp is not None:
-            if brightness is None:
-                brightness = self.brightness
-            if color_temp > COLOR_TEMP_WARM_VS_COLD_WHITE_CUT_OFF:
-                self._bulb.setRgbw(w=brightness)
-            else:
-                self._bulb.setRgbw(w2=brightness)
-            return
 
-        # Show warning if effect set with rgb, brightness, or white level
-        if effect and (brightness or white or rgb):
-            _LOGGER.warning(
-                "RGB, brightness and white level are ignored when"
-                " an effect is specified for a flux bulb"
+        if effect == EFFECT_RANDOM:
+            color_red = random.randint(0, 255)
+            color_green = random.randint(0, 255)
+            color_blue = random.randint(0, 255)
+
+            self._bulb.setRgbw(
+                r=color_red,
+                g=color_green,
+                b=color_blue,
             )
 
-        # Random color effect
-        if effect == EFFECT_RANDOM:
-            self._bulb.setRgb(
-                random.randint(0, 255), random.randint(0, 255), random.randint(0, 255)
+            self._hs_color = color_util.color_RGB_to_hs(
+                color_red,
+                color_green,
+                color_blue,
             )
-            return
 
-        if effect == EFFECT_CUSTOM:
-            if self._custom_effect:
-                self._bulb.setCustomPattern(
-                    self._custom_effect[CONF_COLORS],
-                    self._custom_effect[CONF_SPEED_PCT],
-                    self._custom_effect[CONF_TRANSITION],
-                )
             return
 
-        # Effect selection
         if effect in EFFECT_MAP:
+            self._current_effect = effect
+            self.async_write_ha_state()
             self._bulb.setPresetPattern(EFFECT_MAP[effect], self._effect_speed)
```

Comment:
If this is doing I/O we need to schedule it on the executor.

Reply:
We wrapped the whole _bulb in the executor to handle that

Reply:
What do you mean? Each call that does blocking I/O need to run in the executor.

Normally if the device library doesn't support asyncio its better to use our sync api in the entity actuator methods, ie `turn_on` instead of `async_turn_on`.

Reply:
OK - switched to the sync api.